bool initialized_
duration data